test/encryptedSchema.test.js

```diff
@@ -22,94 +22,174 @@ function schemaHasEncryptedProperty(schema, path) {
+        encryptedFields: {
+          fields: [
+            { path: 'field.encrypted', keyId: KEY_ID, bsonType: 'string' }
+          ]
+        }
+      },
+      'nested object for csfle':
+      {
+        schemaFactory: () => {
+          return new Schema({
+            field: {
+              encrypted: {
+                type: String, encrypt: { keyId: KEY_ID, algorithm }
+              }
+            }
+          }, { encryptionType: 'csfle' });
+        }, predicate: (schema) => assert.ok(schemaHasEncryptedProperty(schema, ['field', 'encrypted'])),
+        schemaMap: {
+          bsonType: 'object',
+          properties: {
+            field: {
+              bsonType: 'object',
+              properties: {
+                encrypted: { encrypt: { bsonType: 'string', algorithm, keyId: KEY_ID } }
+              }
+            }
           }
-        }, { encryptionType: 'csfle' });
-      });
+        }
+      },
+      'nested object for qe': {
+        schemaFactory: () => {
+          return new Schema({
+            field: {
+              encrypted: {
+                type: String, encrypt: { keyId: KEY_ID }
+              }
+            }
+          }, { encryptionType: 'qe' });
+        }, predicate: (schema) => assert.ok(schemaHasEncryptedProperty(schema, ['field', 'encrypted'])),
+        encryptedFields: {
+          fields: [
+            { path: 'field.encrypted', keyId: KEY_ID, bsonType: 'string' }
+          ]
+        }
+      },
+      'schema with encrypted array for csfle': {
+        schemaFactory: () => {
+          return new Schema({
+            encrypted: {
+              type: [Number],
+              encrypt: { keyId: KEY_ID, algorithm }
+            }
+          }, { encryptionType: 'csfle' });
+        }, predicate: (schema) => assert.ok(schemaHasEncryptedProperty(schema, ['encrypted'])),
+        schemaMap: {
+          bsonType: 'object',
+          properties: {
+            encrypted: {
+              encrypt: {
+                bsonType: 'array',
+                keyId: KEY_ID,
+                algorithm
+              }
+            }
+          }
+        }
+      },
+      'schema with encrypted array for qe': {
+        schemaFactory: () => {
+          return new Schema({
+            encrypted: {
+              type: [Number],
+              encrypt: { keyId: KEY_ID }
+            }
+          }, { encryptionType: 'qe' });
+        }, predicate: (schema) => assert.ok(schemaHasEncryptedProperty(schema, ['encrypted'])),
+        encryptedFields: {
+          fields: [
+            { path: 'encrypted', keyId: KEY_ID, bsonType: 'array' }
+          ]
+        }
+      }
+    };
 
-      it('then the schema has a nested property that is encrypted', function() {
-        assert.ok(schemaHasEncryptedProperty(schema, 'encrypted'));
-      });
-    });
+    for (const [description, { schemaFactory, predicate, schemaMap, encryptedFields }] of Object.entries(tests)) {
```

Comment:
I would prefer if you flattened out these tests into distinct `it()` calls. You aren't saving very much SLOC and you're adding some mental overhead for people reviewing this code.

Reply:
Sure, done